### ISSUE TITLE
LazyLoading in ui-router resolve and $compile when done loading 

### ISSUE THREAD

**Issue by maeishoj** [ocombe/ocLazyLoad#209]
Hey there,

I am loading components in my ui-routes. Sometimes I need to load a component like angular-file-upload which I inject in my app, and of course I get "Uncaught Error: [$injector:modulerr] http://errors.angularjs.org/1.3.15/$injector/modulerr?p0=app&p1=Error%3A%20…%2Fmaterial-design-angular%2Fvendors%2Fangular%2Fangular.min.js%3A17%3A381)"

This is what I have:

resolve: {
                    loadPlugin: function($ocLazyLoad) {
                        return $ocLazyLoad.load([
                            {
                                name: 'css',
                                insertBefore: '#app-level',
                                files: [
                                    'js/vendor/material-design-angular/vendors/bower_components/bootstrap-select/dist/css/bootstrap-select.css',
                                    'js/vendor/material-design-angular/vendors/chosen_v1.4.2/chosen.min.css',
                                    'js/vendor/material-design-angular/vendors/bower_components/nouislider/distribute/jquery.nouislider.min.css',
                                    'js/vendor/material-design-angular/vendors/bower_components/eonasdan-bootstrap-datetimepicker/build/css/bootstrap-datetimepicker.min.css',
                                     'assets/css/morph.css'
                                ]
                            },
                            {
                                name: 'vendors',
                                files: [
                                    'js/vendor/material-design-angular/vendors/input-mask/input-mask.min.js',
                                    'js/vendor/material-design-angular/vendors/bower_components/bootstrap-select/dist/js/bootstrap-select.js',
                                    'js/vendor/material-design-angular/vendors/chosen_v1.4.2/chosen.jquery.min.js',
                                    'js/vendor/material-design-angular/vendors/bower_components/nouislider/distribute/jquery.nouislider.all.min.js',
                                    'js/vendor/material-design-angular/vendors/bower_components/moment/min/moment.min.js',
                                    'js/vendor/material-design-angular/vendors/bower_components/eonasdan-bootstrap-datetimepicker/build/js/bootstrap-datetimepicker.min.js',
                                    'js/vendor/material-design-angular/vendors/fileinput/fileinput.min.js',
                                    'js/vendor/components/angular-ui-tree-filter/angular-ui-tree-filter.min.js'
                                ]
                            }
                        ]).then(function(){
                            $compile(angular.element('body'))($scope);
                        })
                    }
                }

Even in the .then and trying to re-compile, does not help.

Anyone can help me out?

**Comment by ocombe** [comment]
Do some of the files that you load have dependencies with each other ?

**Comment by maeishoj** [comment]
No, not really..  if I try to load angular-file-upload only here lazily but not in my index.html I get that error.
And that is cause else when i register modules, at that time it doesn't know angularFileUpload. How can I add the module dynamically instead?

**Comment by ocombe** [comment]
Any chance that you could make me a plunkr with this?

**Comment by maeishoj** [comment]
I can try to do that later, but I wonder if you or someone may tell me how do i actually load a module in my app.module because that is the issue. nothing else.. the reason why it doesn't work is because at runtime, it doesnt know what angularfileupload is, and should not be there. should only be placed in my app.module after I load it.

**Comment by rebelliard** [comment]
I have a similar error, but as OP said, the error is very elusive. I get about 3 notifications on https://getsentry.com about it, but it has never happened to me personally, so it's hard for me to `plunkr` this.

Currently using `1.0.1` on production, and testing out `1.0.5`
